src/field_theory/minimal_polynomial.lean

```diff
@@ -90,7 +97,35 @@ begin
   simpa only [eq_one, alg_hom.map_one, one_ne_zero] using aeval hx
 end
 
-/--A minimal polynomial is not a unit.-/
+/-- If `L/K` is a ring extension, and `x` is an element of `L` in the image of `K`,
+then the minimal polynomial of `x` is `X - C x`. -/
+lemma algebra_map_inj [nontrivial α] (a : α) (hf : function.injective (algebra_map α β))
```

Comment:
This naming surprises me a bit - I'd expect this to be a proof _of_ `function.injective (algebra_map α β)`, not one that takes that as an assumption.

Reply:
Perhaps `eq_if_algebra_map_inj`?

Reply:
Ah, I see this matches `algebra_map` and `algebra_map'` below.

Reply:
I still have some problem with the name convention, so it is probably just my fault...

Reply:
I'm inclined to say that we should fix the other names instead. But I don't have a very good suggestion atm.

Reply:
Maybe something like `of_base_ring`? With the namespace `minimal_polynomial` this would be `minimal.polynomyal.of_base_ring` whose meaning is quite clear.

Reply:
I would expect `minimal.polynomial.of_base_ring` to have type `_  → minimal_polynomial _`, but this has type `_  → minimal_polynomial _ = _` - hence my suggestion `eq_if_algebra_map_inj`.

Whatever name you pick, you should make sure you can rename `algebra_map` and `algebra_map'` into a similar pattern.

Reply:
It seems Kevin's comment about `is_integral α (algebra_map α β a)` applies to `algebra_map` too - it has no need for that condition, right?
```

Reply:
Yes, that's why there is `algebra_map'`. It is exactly `algebra_map` without that assumption. We can remove one of them I think.

Reply:
Just to clarify: I added `inj` in the name to distinguish it from`algebra_map`, but the injectivity, although necessary, does not need to appear in the name. Indeed I don't see any concrete situation where one want to speak about a minimal polynomial without the injectivity condition (for example the minimal polynomial is not even unique in that case... something quite weird)

Reply:
Perhaps these should be:
* `minimal_polynomial.eq_X_sub_C_if_algebra_map_inj` (currently `minimal_polynomial.algebra_map_inj')
* `minimal_polynomial.eq_X_sub_C` (currently `minimal_polynomial.algebra_map')

Reply:
I think I like that (with `_if_` changed to `_of_`).